sdk/core/azure-core/azure/core/tracing/ext/opencensus_wrapper.py

```diff
@@ -0,0 +1,169 @@
+
+    def __init__(self, span=None, name="parent_span"):
+        # type: (Span, str) -> None
+        """
+        If a span is not passed in, creates a new tracer. If the instrumentation key for Azure Exporter is given, will
+        configure the azure exporter else will just create a new tracer.
+
+        :param span: The opencensus span to wrap
+        :param name: The name of the opencensus span to create if a new span is needed
+        """
+        super(OpencensusWrapper, self).__init__(span, name)
+        tracer = self.get_current_tracer()
+        self.was_created_by_azure_sdk = False
+        if span is None and (tracer is None or isinstance(tracer, NoopTracer)):
+            azure_exporter_instrumentation_key = self._get_environ(
+                "APPINSIGHTS_INSTRUMENTATIONKEY"
+            )
+            prob = float(self._get_environ("AZURE_TRACING_SAMPLER") or 0)
+            if azure_exporter_instrumentation_key is not None and AzureExporter:
+                tracer = tracer_module.Tracer(
+                    exporter=AzureExporter(
+                        instrumentation_key=azure_exporter_instrumentation_key
+                    ),
+                    sampler=ProbabilitySampler(prob),
+                )
+            else:
+                tracer = tracer or tracer_module.Tracer(sampler=ProbabilitySampler(prob))
+            self.was_created_by_azure_sdk = True
+        span = span or tracer.span(name=name)
+
+        self.tracer = tracer
+        self._span_instance = span
+
+    @property
+    def span_instance(self):
+        # type: () -> Span
+        """
+        :return: The openencensus span that is being wrapped.
+        """
+        return self._span_instance
+
+    def _get_environ(self, key):
+        # type: (str) -> Union[str, None]
+        """
+        Gets the Environment Variable given. Will return None if no such environment variable is found.
+        :param key: The name of the environment variable
+        :return: The value of the variable or None if the variable does not exist.
+        """
+        if key in os.environ:
+            return os.environ[key]
+        return None
+
+    def span(self, name="child_span"):
+        # type: (str) -> OpencensusWrapper
+        """
+        Create a child span for the current span and append it to the child spans list in the span instance.
+        :param name: Name of the child span
+        :return: The OpencensusWrapper that is wrapping the child span instance
+        """
+        return self.__class__(self.span_instance.span(name=name))
+
+    def start(self):
+        # type: () -> None
+        """Set the start time for a span."""
+        self.span_instance.start()
+
+    def finish(self):
+        # type: () -> None
+        """Set the end time for a span."""
+        self.span_instance.finish()
+        if self.was_created_by_azure_sdk:
+            self.end_tracer(self.tracer)
+
+    def to_header(self, headers):
```

Comment:
Why does `to_header` require headers?

Reply:
some other implementation require it. For example, datadog does not return a new dictionary but modifies the passed in dictionary

Reply:
you are right the wrapper should handle it. Fixed it.